frontends/p4/substitution.h

```diff
@@ -76,21 +76,11 @@ class TypeVariableSubstitution final : public TypeSubstitution<const IR::ITypeVa
  public:
     TypeVariableSubstitution() = default;
     TypeVariableSubstitution(const TypeVariableSubstitution& other) = default;
-    /* Set bindings for multiple variables at the same time.
-     * @param params   List of variables to set.
-     * @param args     List of type arguments to set to variables.
-     * @param root     P4 program element which is being processed. Used for error reporting.
-     * @return         True on success. */
-    bool setBindings(const IR::Node* root,
```

Comment:
Several of the passes that I wrote have a comment describing briefly what they do, often using a simple example. I am not sure any of these IDEs work with our complicated build system.
The basic git usage is fine, and I bet for more complicated stuff the IDEs will fail too.